security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -38,6 +40,12 @@ const (
 
 var (
 	citadelClientLog = log.RegisterScope("citadelclient", "citadel client debugging", 0)
+
+	// ProvCert is the environment controlling the use of pre-provisioned certs, for VMs.
```

Comment:
@costinm could we put in some comments on how the ProvCert certs are built?  Used to be from the istio.default secret, per https://istio.io/docs/examples/virtual-machines/single-network/#preparing-the-kubernetes-cluster-for-vms

Reply:
It was never really 'from istio.default' - that was just a quick example for demo. 

The cert must have a spiffee SAN with the actual trust domain and NAMESPACE where the VM is running the service. And it should be generated using the root CA. 

For quick demo/test:

go run ./security/tools/generate_cert -client -host spiffee://cluster.local/a/b -mode citadel  --out-cert /etc/certs/cert-chain.pem --out-priv /etc/certs/key.pem

For real prod - use Ansible/Puppet/etc, with a root CA that is NOT stored in the cluster, and with Istiod using an intermediate cert from the same root CA. I'm sure there are some good "Enterprise management of certificates" docs and policies we can link to instead of reinventing the wheel :-).

Reply:
@costinm cool, thanks for the clarification.  cc @irisdingbj